lms/models/organization.py

```diff
@@ -46,3 +55,24 @@ class Organization(CreatedUpdatedMixin, PublicIdMixin, BASE):
         nullable=False,
     )
     """Arbitrary settings for the organization."""
+
+    _public_id = sa.Column(
+        "public_id",
+        sa.UnicodeText(),
+        nullable=False,
+        unique=True,
+        default=_generate_public_id,
+    )
+
+    @property
+    def public_id(self):
+        return f"{environ['REGION_CODE']}.lms.org.{self._public_id}"
```

Comment:
It'd be nicer if this could somehow be moved to the services or views layer where `request.registry.settings["region_code"]` is available, to avoid having to read `os.environ` from the models.